apps/meteor/tests/unit/server/services/omnichannel/queue.tests.ts

```diff
@@ -0,0 +1,462 @@
+			models.LivechatInquiry.findNextAndLock.returns(mockedInquiry);
+
+			const queue = new OmnichannelQueue();
+			queue.processWaitingQueue = Sinon.stub().returns(true);
+			queue.execute = Sinon.stub();
+			await queue.checkQueue();
+
+			expect(queue.execute.calledOnce).to.be.true;
+			expect(models.LivechatInquiry.unlock.calledOnce).to.be.true;
+			expect(queue.execute.calledAfter(models.LivechatInquiry.unlock)).to.be.true;
+			expect(queue.execute.calledOnce).to.be.true;
+		});
+	});
+	describe('shouldStart', () => {
+		beforeEach(() => {
+			settings.get.resetHistory();
+			getConfig.resetHistory();
+		});
+		after(() => {
+			settings.get.reset();
+			getConfig.reset();
+		});
+
+		it('should call stop if Livechat is not enabled', async () => {
+			settings.get.returns(false);
+
+			const queue = new OmnichannelQueue();
+			queue.stop = Sinon.stub();
+			await queue.shouldStart();
+
+			expect(queue.stop.calledOnce).to.be.true;
+		});
+		it('should call start if routing algorithm supports auto assignment', async () => {
+			settings.get.returns(true);
+			getConfig.returns({ autoAssignAgent: true });
+
+			const queue = new OmnichannelQueue();
+			queue.start = Sinon.stub();
+			await queue.shouldStart();
+
+			expect(queue.start.calledOnce).to.be.true;
+			expect(queue.start.calledAfter(getConfig)).to.be.true;
+		});
+		it('should call stop if routing algorithm does not support auto assignment', async () => {
+			settings.get.returns(true);
+			getConfig.returns({ autoAssignAgent: false });
+
+			const queue = new OmnichannelQueue();
+			queue.stop = Sinon.stub();
+			await queue.shouldStart();
+
+			expect(queue.stop.calledOnce).to.be.true;
+			expect(queue.stop.calledAfter(getConfig)).to.be.true;
+		});
+	});
+	describe('reconciliation', () => {
+		beforeEach(() => {
+			models.LivechatInquiry.removeByRoomId.resetHistory();
+			models.LivechatInquiry.takeInquiry.resetHistory();
+		});
+
+		it('should remove inquiries from rooms that do not exist', async () => {
+			const queue = new OmnichannelQueue();
+			await queue.reconciliation('missing', { roomId: 'rid', inquiryId: 'inquiryId' });
+
+			expect(models.LivechatInquiry.removeByRoomId.calledOnce).to.be.true;
+		});
+		it('should take an inquiry if the room was taken', async () => {
+			const queue = new OmnichannelQueue();
+			await queue.reconciliation('taken', { roomId: 'rid', inquiryId: 'inquiryId' });
+
+			expect(models.LivechatInquiry.takeInquiry.calledOnce).to.be.true;
+		});
+		it('should remove inquiries from rooms that were closed', async () => {
+			const queue = new OmnichannelQueue();
+			await queue.reconciliation('closed', { roomId: 'rid', inquiryId: 'inquiryId' });
+
+			expect(models.LivechatInquiry.removeByRoomId.calledOnce).to.be.true;
+		});
+		it('should return true for any other case', async () => {
+			const queue = new OmnichannelQueue();
+			expect(await queue.reconciliation('random', { roomId: 'rid', inquiryId: 'inquiryId' })).to.be.true;
```

Comment:
The function returns true no matter what happens. We could assert if the model's functions are not being called.